File: frontend-admin/src/api/file.ts

import request from '@/utils/request'

export interface FileInfo {
  id?: number
  filename: string
  storage_name?: string
  file_path?: string
  file_url?: string
  file_size: number
  file_type?: string
  file_ext?: string
  upload_user_id?: number
  category?: string
  storage_type: string
  status: number
  created_at?: string
  updated_at?: string
}

export interface UploadResponse {
  file_id: number
  filename: string
  file_url: string
  file_size: number
}

export interface PageParams {
  page: number
  page_size: number
}

export interface PageResponse&lt;T&gt; {
  total: number
  page: number
  page_size: number
  items: T[]
}

export interface ApiResponse&lt;T = any&gt; {
  code: number
  message: string
  data: T
}

export const uploadFile = (file: File, category?: string, upload_user_id?: number) =&gt; {
  const formData = new FormData()
  formData.append('file', file)
  if (category) formData.append('category', category)
  if (upload_user_id) formData.append('upload_user_id', String(upload_user_id))
  
  return request.post&lt;ApiResponse&lt;UploadResponse&gt;&gt;('/uploads', formData, {
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

export const getFileList = (params: PageParams &amp; {
  filename?: string
  file_type?: string
  category?: string
  upload_user_id?: number
  status?: number
}) =&gt; {
  return request.get&lt;ApiResponse&lt;PageResponse&lt;FileInfo&gt;&gt;&gt;('/uploads', { params })
}

export const getFileById = (id: number) =&gt; {
  return request.get&lt;ApiResponse&lt;FileInfo&gt;&gt;(`/uploads/${id}`)
}

export const updateFile = (id: number, data: Partial&lt;FileInfo&gt;) =&gt; {
  return request.put&lt;ApiResponse&lt;FileInfo&gt;&gt;(`/uploads/${id}`, data)
}

export const deleteFile = (id: number) =&gt; {
  return request.delete&lt;ApiResponse&lt;any&gt;&gt;(`/uploads/${id}`)
}

export const importExcel = () =&gt; {
  return request.post&lt;ApiResponse&lt;any&gt;&gt;('/uploads/excel/import')
}

export const exportExcel = () =&gt; {
  return request.get&lt;ApiResponse&lt;any&gt;&gt;('/uploads/excel/export')
}

File: backend/app/schemas/file.py

from pydantic import BaseModel, Field
from typing import Optional
from .common import BaseSchema, PageParams


class FileBase(BaseModel):
    filename: str = Field(..., description="原始文件名")
    storage_name: Optional[str] = None
    file_path: Optional[str] = None
    file_url: Optional[str] = None
    file_size: int = Field(default=0, description="文件大小")
    file_type: Optional[str] = None
    file_ext: Optional[str] = None
    upload_user_id: Optional[int] = None
    category: Optional[str] = None
    storage_type: str = Field(default="local", description="存储类型")
    status: int = Field(default=1, description="状态")


class FileCreate(FileBase):
    pass


class FileUpdate(BaseModel):
    filename: Optional[str] = None
    category: Optional[str] = None
    status: Optional[int] = None


class FileResponse(FileBase, BaseSchema):
    pass


class FileQuery(PageParams):
    filename: Optional[str] = None
    file_type: Optional[str] = None
    category: Optional[str] = None
    upload_user_id: Optional[int] = None
    status: Optional[int] = None


class UploadResponse(BaseModel):
    file_id: int = Field(..., description="文件ID")
    filename: str = Field(..., description="文件名")
    file_url: str = Field(..., description="文件URL")
    file_size: int = Field(..., description="文件大小")
